src/Psalm/Internal/Provider/ClassLikeStorageCacheProvider.php

```diff
@@ -111,7 +113,8 @@ public function getLatestFromCache(
 
     private function getCacheHash(?string $file_path, ?string $file_contents): string
     {
-        return sha1(($file_path ? $file_contents : '') . $this->modified_timestamps);
+        $data = ($file_path ? $file_contents : '') . $this->modified_timestamps;
+        return version_compare(phpversion(), '8.1', '>=') ? hash('xxh128', $data) : hash('md4', $data);
```

Comment:
If that's a hot path, a check like `PHP_VERSION_ID > 80100` would make more sense.

Reply:
Thanks for the feedback.
Given the current codebase of psalm, this is quite a microptimization (this change makes the psalm analysis 0.03% faster than version_compare as per profiler).

I'm happy to adjust this though. Can I just add it to the existing PR? or do I need to create a new one?

Reply:
I don't think you can add on this one but maybe add this to the new one you created fo igbinary?